packages/material-ui-lab/src/TreeNode/TreeNode.test.js

```diff
@@ -0,0 +1,555 @@
+        const { getByTestId, getByText } = render(
+          <TreeView>
+            <TreeNode nodeId="1" label="one" data-testid="one" />
+            <TreeNode nodeId="2" label="two" />
+            <TreeNode nodeId="3" label="three" />
+          </TreeView>,
+        );
+
+        fireEvent.focus(getByText('one'));
+        fireEvent.focus(getByText('two'));
+        fireEvent.focus(getByText('three'));
+
+        expect(getByTestId('one')).to.be.focused;
+      });
+
+      it('should focus the selected node if a node is selected before the tree receives focus', () => {
+        const { getByTestId, getByText } = render(
+          <TreeView>
+            <TreeNode nodeId="1" label="one" />
+            <TreeNode nodeId="2" label="two" data-testid="two" />
+            <TreeNode nodeId="3" label="three" />
+          </TreeView>,
+        );
+
+        fireEvent.click(getByText('two'));
+
+        fireEvent.focus(getByText('one'));
+        fireEvent.focus(getByText('two'));
+        fireEvent.focus(getByText('three'));
+
+        expect(getByTestId('two')).to.be.focused;
+      });
+    });
+
+    describe('right arrow interaction', () => {
+      it('should open the node and not move the focus if focus is on a closed node', () => {
+        const { getByTestId, getByText } = render(
+          <TreeView>
+            <TreeNode nodeId="one" label="one" data-testid="one">
+              <TreeNode nodeId="two" label="two" />
+            </TreeNode>
+          </TreeView>,
+        );
+
+        expect(getByTestId('one')).to.have.attribute('aria-expanded', 'false');
+        fireEvent.focus(getByTestId('one'));
+        fireEvent.keyDown(getByText('one'), { key: 'ArrowRight' });
+        expect(getByTestId('one')).to.have.attribute('aria-expanded', 'true');
+        expect(getByTestId('one')).to.be.focused;
+      });
+
+      it('should move focus to the first child if focus is on an open node', () => {
+        const { getByTestId, getByText } = render(
+          <TreeView expanded={['one']}>
+            <TreeNode nodeId="one" label="one" data-testid="one">
+              <TreeNode nodeId="two" label="two" data-testid="two" />
+            </TreeNode>
+          </TreeView>,
+        );
+
+        expect(getByTestId('one')).to.have.attribute('aria-expanded', 'true');
+        fireEvent.focus(getByTestId('one'));
+        fireEvent.keyDown(getByText('one'), { key: 'ArrowRight' });
+        expect(getByTestId('two')).to.be.focused;
+      });
+
+      it('should do nothing if focus is on an end node', () => {
+        const { getByTestId, getByText } = render(
+          <TreeView expanded={['one']}>
+            <TreeNode nodeId="one" label="one" data-testid="one">
+              <TreeNode nodeId="two" label="two" data-testid="two" />
+            </TreeNode>
+          </TreeView>,
+        );
+
+        fireEvent.click(getByText('two'));
```

Comment:
What happens if we actually click here i.e. `getByText('two').click()`

Reply:
The focus doesn't move